code/datums/spells/alien_spells/queen_evolve.dm

```diff
@@ -0,0 +1,27 @@
+/obj/effect/proc_holder/spell/alien_spell/evolve_queen
+	name = "Evolve into an alien queen"
+	desc = "Evolve into an alien queen."
+	plasma_cost = 500
+	action_icon_state = "alien_evolve_drone"
+
+/obj/effect/proc_holder/spell/alien_spell/evolve_queen/create_new_targeting()
+	return new /datum/spell_targeting/self
+
+/obj/effect/proc_holder/spell/alien_spell/evolve_queen/cast(list/targets, mob/living/carbon/user)
+	/// First we check if there is a living queen
+	for(var/mob/living/carbon/alien/humanoid/queen/living_queen in GLOB.alive_mob_list)
+		if(living_queen.key || !living_queen.get_int_organ(/obj/item/organ/internal/brain)) // We do a once over to check the queen didn't end up going away into the magic land of semi-dead
+			to_chat(user, "<span class='notice'>We already have an alive queen.</span>")
+			revert_cast(user)
+			return
+	// If there is no queen, that means we can evolve
+	to_chat(user, "<span class='noticealien'>You begin to evolve!</span>")
+	for(var/mob/O in viewers(user, null))
+		O.show_message(text("<span class='alertalien'>[user] begins to twist and contort!</span>"), 1)
```

Comment:
Can turn this into a `user.visible_message` call

Reply:
Done